ReactAndroid/src/main/java/com/facebook/react/uimanager/animation/NativeAnimatedModule.java

```diff
@@ -0,0 +1,228 @@
+    // use
+    ArrayList<UIThreadOperation> operations = mOperations.isEmpty() ? null : mOperations;
+    if (operations != null) {
+      mOperations = new ArrayList<>();
+      synchronized (mOperationsCopyLock) {
+        if (mReadyOperations == null) {
+          mReadyOperations = operations;
+        } else {
+          mReadyOperations.addAll(operations);
+        }
+      }
+    }
+  }
+
+  @Override
+  public void onHostResume() {
+    enqueueFrameCallback();
+  }
+
+  @Override
+  public void onHostPause() {
+    clearFrameCallback();
+  }
+
+  @Override
+  public void onHostDestroy() {
+    clearFrameCallback();
+  }
+
+  @Override
+  public String getName() {
+    return "NativeAnimatedModule";
+  }
+
+  private void clearFrameCallback() {
+    Assertions.assertNotNull(mReactChoreographer).removeFrameCallback(
+      ReactChoreographer.CallbackType.ANIMATIONS,
+      mAnimatedFrameCallback);
+  }
+
+  private void enqueueFrameCallback() {
+    Assertions.assertNotNull(mReactChoreographer).postFrameCallback(
+      ReactChoreographer.CallbackType.ANIMATIONS,
+      mAnimatedFrameCallback);
+  }
+
+  @ReactMethod
+  public void createAnimatedNode(final int tag, final ReadableMap config) {
+    mOperations.add(new UIThreadOperation() {
+      @Override
+      public void execute(NativeAnimatedNodesManager animatedNodesManager) {
+        animatedNodesManager.createAnimatedNode(tag, config);
+      }
+    });
+  }
+
+  @ReactMethod
+  public void dropAnimatedNode(final int tag) {
+    mOperations.add(new UIThreadOperation() {
+      @Override
+      public void execute(NativeAnimatedNodesManager animatedNodesManager) {
+        animatedNodesManager.dropAnimatedNode(tag);
+      }
+    });
+  }
+
+  @ReactMethod
+  public void setAnimatedNodeValue(final int tag, final double value) {
+    mOperations.add(new UIThreadOperation() {
+      @Override
+      public void execute(NativeAnimatedNodesManager animatedNodesManager) {
+        animatedNodesManager.setAnimatedNodeValue(tag, value);
+      }
+    });
+  }
+
+  @ReactMethod
+  public void startAnimatingNode(
+      final int animatedNodeTag,
+      final ReadableMap animationConfig,
+      final Callback endCallback) {
+    mOperations.add(new UIThreadOperation() {
+      @Override
+      public void execute(NativeAnimatedNodesManager animatedNodesManager) {
+        animatedNodesManager.startAnimatingNode(
+          animatedNodeTag,
+          animationConfig,
+          endCallback);
+      }
+    });
+  }
+
+  @ReactMethod
+  public void connectAnimatedNodes(final int parentNodeTag, final int childNodeTag) {
+    mOperations.add(new UIThreadOperation() {
+      @Override
+      public void execute(NativeAnimatedNodesManager animatedNodesManager) {
+        animatedNodesManager.connectAnimatedNodes(parentNodeTag, childNodeTag);
+      }
+    });
+  }
+
+  @ReactMethod
+  public void disconnectAnimatedNodes(final int parentNodeTag, final int childNodeTag) {
```

Comment:
not really, it's just that I cannot directly map JS attach calls to native attach calls (nodes can be build separately and only when start is called I know if the graph should be sent to native)